lib/spack/spack/cmd/checksum.py

```diff
@@ -33,35 +36,38 @@ def setup_parser(subparser):
         default=False,
         help="don't clean up staging area when command completes",
     )
-    sp = subparser.add_mutually_exclusive_group()
-    sp.add_argument(
+    subparser.add_argument(
```

Comment:
In refactoring checksum I couldn't come up with a great reason why `-l`, and `-p` should be mutually exclusive and thus decided to refactor them so they could be used together.

For example let's say we want to get 3 version checksums of `Go`,
1. The latest available version. `v1.21rc2`
2. The preferred version in Spack. `v1.20.4`
3. And `v1.20.5`

Where in the previous version of Spack this took multiple commands, you can now do it all with
```
> spack checksum -l -p go 1.20.5

==> Found 3 versions of go:

  1.21rc2  https://go.dev/dl/go1.21rc2.src.tar.gz
  1.20.5   https://go.dev/dl/go1.20.5.src.tar.gz
  1.20.4   https://go.dev/dl/go1.20.4.src.tar.gz

==> Fetching https://go.dev/dl/go1.21rc2.src.tar.gz
==> Fetching https://go.dev/dl/go1.20.5.src.tar.gz
==> Fetching https://go.dev/dl/go1.20.4.src.tar.gz

    version("1.21rc2", sha256="91596dce22d91f2d7be6d2b49829f39bdd8573e777d95c1b3c6b0d42b613076e")
    version("1.20.5", sha256="9a15c133ba2cfafe79652f4815b62e7cfc267f68df1b9454c6ab2a3ca8b96a88")
    version("1.20.4", sha256="9f34ace128764b7a3a4b238b805856cc1b2184304df9e5690825b0710f4202d6")
```

`-b` `--batch` is a little more confusing since it doesn't actually make a difference unless you haven't given any versions at all at which it's used to skip the interactive prompt. We should probably rename `--batch` to `--yes` to standardize on flags, but I thought that might be out of the scope of this PR. In any case, I didn't think a user should be punished (have an mutex error be thrown) if they were to pass `-b` with `-l` or `-p` since it wouldn't affect the output.

If any of this could be made better though I'm all ears! Thanks for reviewing @tldahlgren!